lib/system/excpt.nim

```diff
@@ -647,7 +647,17 @@ when not defined(noSignalHandler) and not defined(useNimRtl):
       # unless there's a good reason to use cstring in signal handler to avoid
       # using gc?
       showErrorMessage(msg, msg.len)
-    quit(1) # always quit when SIGABRT
+
+    when defined(posix):
+      # reset the signal handler to OS default
+      c_signal(sign, SIG_DFL)
+
+      # re-raise the signal, which will arrive once this handler exit.
+      # this lets the OS perform actions like core dumping and will
+      # also return the correct exit code to the shell.
+      discard c_raise(sign)
+    else:
+      quit(1)
```

Comment:
```suggestion
    quit(1)
```

so that even if c_raise fails quit is still called

Reply:
The signal itself is blocked in the signal handler, so raising it will not cause a crash until the handler exit.

We can unblock the signal, but it would require introducing too many duplicate declarations with `posix`. Instead I just exit the handler, because `c_raise()` can only fail if the signal is invalid, and since we are handling the said signal, it can't be invalid.

Reply:
* can you write this as comment in the code?
* also, what about race conditions?

Reply:
> re-raise the signal, which will arrive once this handler exit.

I did write this, but I guess it wasn't obvious enough.

> also, what about race conditions?

What race conditions? It doesn't matter who raised the signal, the default handler will always crash the process. And since signals are queued, it will always end up in the signal that we raise, assuming that the other signals didn't crash the program.